src/Components/Components/src/IComponentActivator.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Represents an activator that can be used to instantiate components.
+    /// </summary>
+    public interface IComponentActivator
+    {
+        /// <summary>
+        /// Creates an component of the specified type using that type's default constructor.
+        /// </summary>
+        /// <param name="componentType">The type of component to create.</param>
+        /// <returns>A reference to the newly created component.</returns>
+        object CreateInstance(Type componentType);
```

Comment:
```suggestion
        IComponent CreateInstance(Type componentType);
```
Should this return an `IComponent` instead of just object? It seems that all components are `IComponent`s, and having this returned here might save a cast higher up in the call stack.

Reply:
You're probably right, but I just wanted to make it behave the same as `Activator.CreateInstance(componentType)` which is returning an `object`.

Reply:
Since we require component instances to implement `IComponent`, I think it only makes sense if the activator returns that type.